internal/command/jsonformat/change/object.go

```diff
@@ -0,0 +1,88 @@
+package change
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+
+	"github.com/hashicorp/terraform/internal/command/format"
+	"github.com/hashicorp/terraform/internal/plans"
+)
+
+func Object(attributes map[string]Change) Renderer {
+	maximumKeyLen := 0
+	for key := range attributes {
+		if maximumKeyLen < len(key) {
+			maximumKeyLen = len(key)
+		}
+	}
+
+	return &objectRenderer{
+		attributes:         attributes,
+		maximumKeyLen:      maximumKeyLen,
+		overrideNullSuffix: true,
+	}
+}
+
+func NestedObject(attributes map[string]Change) Renderer {
+	maximumKeyLen := 0
+	for key := range attributes {
+		if maximumKeyLen < len(key) {
+			maximumKeyLen = len(key)
+		}
+	}
+
+	return &objectRenderer{
+		attributes:         attributes,
+		maximumKeyLen:      maximumKeyLen,
+		overrideNullSuffix: false,
+	}
+}
+
+type objectRenderer struct {
+	NoWarningsRenderer
+
+	attributes         map[string]Change
+	maximumKeyLen      int
+	overrideNullSuffix bool
+}
+
+func (renderer objectRenderer) Render(change Change, indent int, opts RenderOpts) string {
+	if len(renderer.attributes) == 0 {
+		return fmt.Sprintf("{}%s%s", change.nullSuffix(opts.overrideNullSuffix), change.forcesReplacement())
+	}
+
+	attributeOpts := opts.Clone()
+	attributeOpts.overrideNullSuffix = renderer.overrideNullSuffix
+
+	var keys []string
+	for key := range renderer.attributes {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	unchangedAttributes := 0
+	var buf bytes.Buffer
+	buf.WriteString(fmt.Sprintf("{%s\n", change.forcesReplacement()))
+	for _, key := range keys {
+		attribute := renderer.attributes[key]
+
+		if attribute.action == plans.NoOp && !opts.showUnchangedChildren {
+			// Don't render NoOp operations when we are compact display.
+			unchangedAttributes++
+			continue
+		}
+
+		for _, warning := range attribute.Warnings(indent + 1) {
+			buf.WriteString(fmt.Sprintf("%s%s\n", change.indent(indent+1), warning))
```

Comment:
The `indent + 1` vs `indent+1` difference here confuses me. Why doesn't `gofmt` make these consistent?

Reply:
I hadn't noticed this, but it is apparently deliberate by gofmt: https://stackoverflow.com/a/59024523